astropy/table/table.py

```diff
@@ -3652,28 +3652,26 @@ def _encode_mixins(tbl):
         out = OrderedDict()
 
         for name, column in tbl.columns.items():
+            if getattr(column.dtype, 'isnative', True):
+                out[name] = column
+            else:
+                out[name] = column.data.byteswap().newbyteorder('=')
+
             if isinstance(column, MaskedColumn) and np.any(column.mask):
                 if column.dtype.kind in ['i', 'u']:
-                    pd_dtype = str(column.dtype)
+                    pd_dtype = column.dtype.name
                     if use_nullable_int:
                         # Convert int64 to Int64, uint32 to UInt32, etc for nullable types
-                        pd_dtype = pd_dtype.replace('i', 'I').replace('u', 'U')
-                    out[name] = Series(column, dtype=pd_dtype)
+                        pd_dtype = pd_dtype.capitalize().replace('i', 'I')
```

Comment:
Another nitpick. I like the original version of this line as being more consistent, as in capitalize any `i` or `u` in `pd_dtype` using `replace`. If there is nothing logically wrong with the original then please revert.

Reply:
Sure; I just wanted to use `capitalize` as a neater method, but then realised that would not be enough to deal with the `uint`s.